pkg/algorithm/cstorpoolselect/v1alpha1/select_test.go

```diff
@@ -157,3 +165,152 @@ func TestNewSelection(t *testing.T) {
 		})
 	}
 }
+
+func TestAntiAffinityFilter(t *testing.T) {
+	tests := map[string]struct {
+		CVRUid, poolUids, expectedUids []string
+		labelSelector                  string
+		expectError                    bool
+	}{
+		"Test 1":  {[]string{}, []string{"uid 1", "uid 2", "uid 3"}, []string{}, "label3", true},
+		"Test 2":  {[]string{"uid 4", "uid 2", "uid 7"}, []string{"uid 6"}, []string{}, "label1", true},
+		"Test 3":  {[]string{"uid 4", "uid 2", "uid 7"}, []string{"uid 6"}, []string{"uid 6"}, "label1", false},
+		"Test 4":  {[]string{"uid 1", "uid 2"}, []string{"uid 1", "uid 2", "uid 3"}, []string{}, "label2", true},
+		"Test 5":  {[]string{"uid 1", "uid 2"}, []string{"uid 1", "uid 2", "uid 3"}, []string{"uid 3"}, "label2", false},
+		"Test 6":  {[]string{"uid 1", "uid 2", "uid 3"}, []string{"uid 1", "uid 5", "uid 3"}, []string{}, "label4", true},
+		"Test 7":  {[]string{}, []string{"uid 1", "uid 2", "uid 3"}, []string{"uid 1", "uid 2", "uid 3"}, "label3", false},
+		"Test 8":  {[]string{"uid 1", "uid 2", "uid 3"}, []string{"uid 1", "uid 5", "uid 3"}, []string{"uid 5"}, "label4", false},
+		"Test 9":  {[]string{"uid 1", "uid 2", "uid 3", "uid 4"}, []string{"uid 1", "uid 2", "uid 3", "uid 5"}, []string{}, "label6", true},
+		"Test 10": {[]string{"uid 1", "uid 2", "uid 3", "uid 4"}, []string{"uid 1", "uid 2", "uid 3", "uid 5"}, []string{"uid 5"}, "label5", false},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			fakeList := &apis.CStorVolumeReplicaList{}
```

Comment:
It would have been ideal to build a list of cstor volume replica list using builder functions in `pkg/cstorvolumereplica/v1alpha1`.
Please mark this as a //TODO item.

Reply:
we cannot add label then